scarlet2/initialization.py

```diff
@@ -0,0 +1,331 @@
+            converged = True
+        # increase box/morpgology size
+        else:
+            spectra_prev = spectra_box
+            sigma += 1
+        # for debugging
+        if sigma > 5:
+            break
+
+    # normalise the morphology between 0 and 1
+    morph = morph()
+    morph = (morph - jnp.min(morph)) / (jnp.max(morph) - jnp.min(morph))
+    return morph
+
+
+def init_morphology(
+    obs,
+    center,
+    psf_sigma=1,
+    noise_thresh=100,
+    corr_thresh=0.8,
+    max_size=32,
+    components=1,
+):
+    """Initialize the morphology of the sources.
+    Parameters
+    ----------
+    obs : `~scarlet.observation.Observation` instance
+    center : tuple (int, int) central pixel of the sources
+
+    Returns
+    -------
+    morph : np.ndarray morphology of the source normalised to (0,1)
+    """
+    morph = init_simple_morph(obs, center, psf_sigma, noise_thresh, corr_thresh)
+    if morph.shape[0] <= max_size:
+        return morph
+    # fit for a more complex morphology for bigger sources
+    else:
+        bx = morph.shape[0]
+        cutout = cut_square_box(obs.data[0], center, bx)
+        sx, sy, th = fit_morph_params(cutout, center, bx)
+        morph = create_gaussian_array(bx, sx, sy, th)
+        morph = np.array(morph, dtype=np.float32)  # convert to numpy to make mutable
+        rows, cols = morph.shape
+        central_row = rows // 2
+        central_col = cols // 2
+        morph[central_row, central_col] = 1.1  # extra brightening central pixel
+        morph = (morph - np.min(morph)) / (np.max(morph) - np.min(morph))
+        if bx > 30 and components == 2:
+            morph2 = create_gaussian_array(
+                bx, 1.25, 1.25, 0
+            )  # create a second component as a gaussian blob
+            morph2 = (morph2 - np.min(morph2)) / (np.max(morph2) - np.min(morph2))
+            return [morph, morph2]
+        else:
+            # if single component just return the fitted morhpology
+            return morph
+
+
+# initialise the spectrum
+def init_spectrum(observations, center, correct_psf=None, models=None):
+    """Get the spectrum at center of observation.
+
+    Yields the spectrum of a single-pixel source with flux 1 in every channel,
+    concatenated for all observations.
+
+    If `correct_psf`, it homogenizes the PSFs of the observations, which yields the
+    correct spectrum for a flux=1 point source.
+
+    If `model` is set, it reads of the value of the model at `sky_coord` and yields the
+    spectrum for that model.
+
+    Parameters
+    ----------
+    center: tuple
+        Position in the observation
+    observations: instance or list of `~scarlet.Observation`
+        Observation to extract SED from.
+    correct_psf: bool
+        If PSF shape variations in the observations should be corrected.
+    models: instance or list of arrays
+        Rendered models for this source in every observation
```

Comment:
I think provinding the model is a scarlet1 option we should not longer support. Suggest to remove in this function

Reply:
Removed